aws-android-sdk-mobile-client/src/main/java/com/amazonaws/mobile/client/AWSMobileClient.java

```diff
@@ -1611,6 +1611,9 @@ public Void run() throws Exception {
                     userpoolLL.globalSignOut(globalSignOutRequest);
                 }
                 if (signOutOptions.isInvalidateTokens()) {
+                    if (userpool != null) {
```

Comment:
isn't signOutOptions.invalidateTokens only used for hostedUI? what is the default value? if it is always false when options is not passed in, then the new behavior on line 1615 `userpool.getCurrentUser().revokeTokens();` will never be called. revoke tokens should always be called 
1. when user is signed in with user pools
2. when user is signed in with hostedUI and signOutOptions.invalidateTokens is true

revoke tokens should not be called for federated sign-ins

Reply:
Changed the default value of `invalidateTokens` to true. 

- [x] revoke tokens should always be called when user is signed in with user pools
- [x] revoke tokens should always be called when user is signed in with hostedUI and signOutOptions.invalidateTokens is true
- [ ] revoke tokens should not be called for federated sign-ins.

Reply:
Maybe we need to move lines 1414-1416 outside of `if (signOutOptions.isInvalidateTokens()) {` and refactor the if statement to something like this:

```
if ((hostedUI != null && signOutOptions.isInvalidateTokens()) ||
      (hostedUI == null && userpool != null)) {
          userpool.getCurrentUser().revokeTokens();
      }
}
```
That way, you don't have to change the default value of `invalidateTokens` and we wouldn't have to worry a change in behavior. Let me know if this makes sense.

Reply:
Turns out we don't need to change the default value of `invalidateTokens` since we have documented this in the SDK docs to do that manually. 

- User Pools
Amplify: Enforced by passing invalidateTokens(true) in AWSCognitoAuthPlugin.signOutLocally
SDK: Needs docs update similar to [SDK hostedUI](https://docs.amplify.aws/sdk/auth/hosted-ui/q/platform/android#sign-out-from-hostedui-1).

- HostedUI
Amplify: Enforced by passing invalidateTokens(true) in AWSCognitoAuthPlugin.signOutLocally
SDK: Manually enforced refer [docs](https://docs.amplify.aws/sdk/auth/hosted-ui/q/platform/android#sign-out-from-hostedui-1)

- Federation
Amplify: Only available through hostedUI.
SDK: Not enforced as expected. Users call `AWSMobileClient.getInstance().signOut()`.